src/mlpack/core/optimizers/adam/nadam_update.hpp

```diff
@@ -0,0 +1,160 @@
+/**
+ * @file nadam_update.hpp
+ * @author Sourabh Varshney
+ *
+ * Nadam update rule. Nadam is an optimizer that combines the effect of Adam and 
+ * NAG to the gradient descent to improve its Performance.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_CORE_OPTIMIZERS_ADAM_NADAM_UPDATE_HPP
+#define MLPACK_CORE_OPTIMIZERS_ADAM_NADAM_UPDATE_HPP
+
+#include <mlpack/prereqs.hpp>
+
+namespace mlpack {
+namespace optimization {
+
+/**
+ * Nadam is an optimizer that combines the Adam and NAG.
+ *
+ * For more information, see the following.
+ *
+ * @code
+ * @misc{
+ *   author  = {},
+ *   title   = {},
+ *   journal = {},
+ *   year    = {},
+ *   url     = {}
+ * }
+ * @endcode
+ */
+class NadamUpdate
+{
+ public:
+  /**
+   * Construct the Nadam update policy with the given parameters.
+   *
+   * @param epsilon The epsilon value used to initialise the squared gradient
+   *        parameter.
+   * @param beta1 The smoothing parameter.
+   * @param beta2 The second moment coefficient
+   */
+  NadamUpdate(const double epsilon = 1e-8,
+              const double beta1 = 0.9,
+              const double beta2 = 0.99)
+              :epsilon(epsilon), beta1(beta1), beta2(beta2), iteration(0),
+              cum_beta1(1)
+  {
+    // Nothing to do.
+  }
+
+  /**
+   * The Initialize method is called by SGD Optimizer method before the start 
+   * of the iteration update process.
+   *
+   * @param rows Number of rows in the gradient matrix.
+   * @param cols Number of columns in the gradient matrix.
+   */
+  void Initialize(const size_t rows, const size_t cols)
+  {
+    m = arma::zeros<arma::mat>(rows, cols);
+    v = arma::zeros<arma::mat>(rows, cols);
+  }
+
+  /**
+   * Update step for Nadam.
+   *
+   * @param iterate Parameters that minimize the function.
+   * @param stepSize Step size to be used for the given iteration.
+   * @param gradient The gradient matrix.
+   */
+  void Update(arma::mat& iterate,
+              const double stepSize,
+              const arma::mat& gradient)
+  {
+    // Increment the iteration counter variable.
+    ++iteration;
+
+    // And update the iterate.
+    m *= beta1;
+    m += (1 - beta1) * gradient;
+
+    v *= beta2;
+    v += (1 - beta2) * gradient % gradient;
+
+    // beta1_t = beta1 * (1 - (0.5 * (0.96 ^ (iteration / 250))))
+    double beta1_t = beta1 * (1 - (0.5 * std::pow(0.96, (iteration / 250))));
```

Comment:
What do you think about introducing another parameter for the momentum schedule, instead of using a const one? We could call it ``scheduleDecay`` and set a default value of ``4e-3`` and change the line to:

```
std::pow(0.96, iteration  * scheduleDecay)
```

Also, can you use camel casing for all names (``beta1_t, beta1_t1``); for more information take a look at: https://github.com/mlpack/mlpack/wiki/DesignGuidelines#naming-conventions.

Reply:
I forgot that decay rate should be defined by function call and should not be hard-quoted. I have made the changes.